### PR TITLE
Add performance profiling tool to enable programmatic/interactive/automatic PT-XLA profile capturing

### LINKED ISSUE #3471: Develop a profiler tool that enables "programmable" and "interactive"  profile capture
Develop a profiler tool that enables "programmable" and "ineteractive" profile capture. This work is already implemented by @thisisalbertliang as part of his internship (see [PR](https://github.com/thisisalbertliang/training/pull/24)). It'd be useful to make it available to PyTorch/XLA clients directly.

@thisisalbertliang let's start by adding your work to the `scripts` directory. When porting this work to PyTorch/XLA, please add the corresponding documentation to `capture_profile.py`.

### PR DESCRIPTION
**Pull request by thisisalbertliang**
Resolves #3471 

## Pull Request Summary

This PR adds a performance profiling utility script (`capture_profile.py`) that enables interactive and/or automatic PT-XLA profile capturing from the command line. The motivation behind `capture_profile.py` is to provide PT-XLA users with an easy and simple way to re-use `torch_xla.debug.profiler.trace` for tracing their training jobs.

## How to Run

Instructions on how to run `capture_profile.py` are already clearly documented in the script itself.

Here are some example run commands
```bash
python3 capture_profile.py --service_addr "localhost:9001" --logdir "gs://path/to/logdir" --duration_ms 20000 --interactive loop
```
```bash
python3 capture_profile.py --service_addr "10.0.0.2:9001" --logdir "gs://path/to/logdir" --duration_ms 30000 --automatic 100 60
```

## For Googlers

See b/226973507 for more details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -338,6 +338,7 @@ With PyTorch/XLA we provide a set of performance profiling tooling and auto-metr
 * [Official tutorial](https://cloud.google.com/tpu/docs/pytorch-xla-performance-profiling-tpu-vm)
 * [Colab notebook](https://colab.research.google.com/github/pytorch/xla/blob/master/contrib/colab/pytorch-xla-profiling-colab.ipynb)
 * [Sample MNIST training script with profiling](https://github.com/pytorch/xla/blob/master/test/test_profile_mp_mnist.py)
+* [Utility script for capturing performance profiles](https://github.com/pytorch/xla/blob/master/scripts/capture_profile.py)
```
Comment (miladm) [inline README.md:341]: Please add a reference on how to load the profiled filed on `tensorboard`.

Comment (miladm) [reply README.md:341]: I suggest we port some of the comments in this [README](https://github.com/thisisalbertliang/training/pull/24/files) to PyTorch/XLA. I wonder if the `README` is the best place. Alternatively, we can consider another documentation guide on "Profiling" in PyTorch/XLA.

@JackCaoG wdyt?

Comment (thisisalbertliang) [reply README.md:341]: @miladm 

I am not sure where to add the tensorboard instructions. It seems like the [Official tutorial](https://cloud.google.com/tpu/docs/pytorch-xla-performance-profiling-tpu-vm) and [Colab notebook](https://colab.research.google.com/github/pytorch/xla/blob/master/contrib/colab/pytorch-xla-profiling-colab.ipynb) links in the README already teaches how to use tensorboard anyway?

For now, I have added instructions on how to run tensorboard in the module docstring for `capture_profile.py`. Let me know if you think there is a better location for these instructions.

**Review by miladm — APPROVED** [review]
Thanks @thisisalbertliang 

LGTM

**Comment by thisisalbertliang** [comment]
@miladm 

I just reformatted `capture_profile.py` using `yapf`. The linter test should pass now.

Let me know if I should go ahead and merge this PR.

**Comment by miladm** [comment]
Great. Thanks @thisisalbertliang. 
All tests need to pass before merging - even though your changes are in python. 
I think rerunning the build or pushing an empty commit triggers the the process.

**Comment by JackCaoG** [comment]
@miladm found this pr while writing 1.12 release note. We should add this to the TroubleShotting doc

**Comment by thisisalbertliang** [comment]
> @miladm found this pr while writing 1.12 release note. We should add this to the TroubleShotting doc

thanks @JackCaoG . Just made a quick PR to add `capture_profile.py` to `TROUBLESHOOTING.md`. lmk if this what you wanted.

PR link: https://github.com/pytorch/xla/pull/3670

**Pull request merged**